### ISSUE TITLE
Documentation problems with references

### ISSUE THREAD

**Issue by hbushouse** [spacetelescope/jwst#2287]
There are a few sets of docs that are giving build warnings about undefined labels and more than one target found for cross-reference. These should be fixed by the appropriate responsible parties.

The warnings I've seen are for ``associations``:
```
/Users/bushouse/JWST/softw/jwst/jwst/associations/registry.py:docstring of jwst.associations.AssociationRegistry:53: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/jwst/associations/registry.py:docstring of jwst.associations.AssociationRegistry:53: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/jwst/associations/generate.py:docstring of jwst.associations.generate:18: WARNING: undefined label: association-generator (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/jwst/associations/generate.py:docstring of jwst.associations.generate:18: WARNING: undefined label: association-generator (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:26: WARNING: undefined label: level2-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:26: WARNING: undefined label: level2-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:26: WARNING: undefined label: level3-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:26: WARNING: undefined label: level3-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:40: WARNING: undefined label: main (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:40: WARNING: undefined label: main (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:40: WARNING: undefined label: generate_function (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:40: WARNING: undefined label: generate_function (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:40: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:40: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:40: WARNING: undefined label: asn-registry (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/asn_generate.rst:40: WARNING: undefined label: asn-registry (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association.rst:9: WARNING: undefined label: what-are-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association.rst:9: WARNING: undefined label: what-are-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association.rst:66: WARNING: undefined label: level2-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association.rst:66: WARNING: undefined label: level2-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association.rst:66: WARNING: undefined label: level3-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association.rst:66: WARNING: undefined label: level3-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association.rst:85: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association.rst:85: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:8: WARNING: undefined label: association (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:8: WARNING: undefined label: association (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:13: WARNING: undefined label: association (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:13: WARNING: undefined label: association (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:27: WARNING: undefined label: level3-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:27: WARNING: undefined label: level3-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:128: WARNING: undefined label: asn-registry (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:128: WARNING: undefined label: asn-registry (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:181: WARNING: undefined label: asn-registry (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/association_reference.rst:181: WARNING: undefined label: asn-registry (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/design.rst:14: WARNING: undefined label: overview (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/design.rst:14: WARNING: undefined label: overview (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:16: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:16: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:47: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:47: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:56: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:56: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:56: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:56: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:72: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/generator.rst:72: WARNING: undefined label: generate (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/jwst_conventions.rst:45: WARNING: undefined label: asn-dms-naming (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/jwst_conventions.rst:45: WARNING: undefined label: asn-dms-naming (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/level2_asn_technical.rst:80: WARNING: undefined label: level3-asn-association-types (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/level2_asn_technical.rst:80: WARNING: undefined label: level3-asn-association-types (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/level2_asn_technical.rst:83: WARNING: undefined label: asn-dms-naming (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/level2_asn_technical.rst:83: WARNING: undefined label: asn-dms-naming (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/level3_asn_technical.rst:78: WARNING: undefined label: level3-asn-association-types (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/level3_asn_technical.rst:78: WARNING: undefined label: level3-asn-association-types (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/level3_asn_technical.rst:81: WARNING: undefined label: asn-dms-naming (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/level3_asn_technical.rst:81: WARNING: undefined label: asn-dms-naming (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:50: WARNING: undefined label: association-generator (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:50: WARNING: undefined label: association-generator (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:50: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:50: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:50: WARNING: undefined label: asn-registry (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:50: WARNING: undefined label: asn-registry (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:50: WARNING: undefined label: association (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:50: WARNING: undefined label: association (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:58: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:58: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:58: WARNING: undefined label: level 2 (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:58: WARNING: undefined label: level 2 (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:58: WARNING: undefined label: level2-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:58: WARNING: undefined label: level2-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:58: WARNING: undefined label: level3-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:58: WARNING: undefined label: level3-associations (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:98: WARNING: undefined label: jwst data model (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/overview.rst:98: WARNING: undefined label: jwst data model (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/pool.rst:10: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/pool.rst:10: WARNING: undefined label: asn-pool (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/technote_sdp_workflow.rst:10: WARNING: undefined label: level3-asn-jwst-overview (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/associations/technote_sdp_workflow.rst:10: WARNING: undefined label: level3-asn-jwst-overview (if the link has no caption the label must precede a section header)
```

and ``transforms``:
```
/Users/bushouse/JWST/softw/jwst/jwst/transforms/models.py:docstring of jwst.transforms.IdealToV2V3.evaluate:: WARNING: more than one target found for cross-reference 'v2ref': jwst.transforms.models.V2V3ToIdeal.v2ref, jwst.transforms.IdealToV2V3.v2ref, jwst.transforms.tpcorr.TPCorr.v2ref, jwst.transforms.models.IdealToV2V3.v2ref, jwst.transforms.TPCorr.v2ref, jwst.transforms.V2V3ToIdeal.v2ref
/Users/bushouse/JWST/softw/jwst/jwst/transforms/models.py:docstring of jwst.transforms.IdealToV2V3.evaluate:: WARNING: more than one target found for cross-reference 'v2ref': jwst.transforms.models.V2V3ToIdeal.v2ref, jwst.transforms.IdealToV2V3.v2ref, jwst.transforms.tpcorr.TPCorr.v2ref, jwst.transforms.models.IdealToV2V3.v2ref, jwst.transforms.TPCorr.v2ref, jwst.transforms.V2V3ToIdeal.v2ref
/Users/bushouse/JWST/softw/jwst/jwst/transforms/models.py:docstring of jwst.transforms.V2V3ToIdeal.evaluate:: WARNING: more than one target found for cross-reference 'v2ref': jwst.transforms.models.V2V3ToIdeal.v2ref, jwst.transforms.IdealToV2V3.v2ref, jwst.transforms.tpcorr.TPCorr.v2ref, jwst.transforms.models.IdealToV2V3.v2ref, jwst.transforms.TPCorr.v2ref, jwst.transforms.V2V3ToIdeal.v2ref
/Users/bushouse/JWST/softw/jwst/jwst/transforms/models.py:docstring of jwst.transforms.V2V3ToIdeal.evaluate:: WARNING: more than one target found for cross-reference 'v2ref': jwst.transforms.models.V2V3ToIdeal.v2ref, jwst.transforms.IdealToV2V3.v2ref, jwst.transforms.tpcorr.TPCorr.v2ref, jwst.transforms.models.IdealToV2V3.v2ref, jwst.transforms.TPCorr.v2ref, jwst.transforms.V2V3ToIdeal.v2ref
/Users/bushouse/JWST/softw/jwst/jwst/transforms/models.py:docstring of jwst.transforms.models.IdealToV2V3.evaluate:: WARNING: more than one target found for cross-reference 'v2ref': jwst.transforms.models.V2V3ToIdeal.v2ref, jwst.transforms.IdealToV2V3.v2ref, jwst.transforms.tpcorr.TPCorr.v2ref, jwst.transforms.models.IdealToV2V3.v2ref, jwst.transforms.TPCorr.v2ref, jwst.transforms.V2V3ToIdeal.v2ref
/Users/bushouse/JWST/softw/jwst/jwst/transforms/models.py:docstring of jwst.transforms.models.IdealToV2V3.evaluate:: WARNING: more than one target found for cross-reference 'v2ref': jwst.transforms.models.V2V3ToIdeal.v2ref, jwst.transforms.IdealToV2V3.v2ref, jwst.transforms.tpcorr.TPCorr.v2ref, jwst.transforms.models.IdealToV2V3.v2ref, jwst.transforms.TPCorr.v2ref, jwst.transforms.V2V3ToIdeal.v2ref
/Users/bushouse/JWST/softw/jwst/jwst/transforms/models.py:docstring of jwst.transforms.models.V2V3ToIdeal.evaluate:: WARNING: more than one target found for cross-reference 'v2ref': jwst.transforms.models.V2V3ToIdeal.v2ref, jwst.transforms.IdealToV2V3.v2ref, jwst.transforms.tpcorr.TPCorr.v2ref, jwst.transforms.models.IdealToV2V3.v2ref, jwst.transforms.TPCorr.v2ref, jwst.transforms.V2V3ToIdeal.v2ref
/Users/bushouse/JWST/softw/jwst/jwst/transforms/models.py:docstring of jwst.transforms.models.V2V3ToIdeal.evaluate:: WARNING: more than one target found for cross-reference 'v2ref': jwst.transforms.models.V2V3ToIdeal.v2ref, jwst.transforms.IdealToV2V3.v2ref, jwst.transforms.tpcorr.TPCorr.v2ref, jwst.transforms.models.IdealToV2V3.v2ref, jwst.transforms.TPCorr.v2ref, jwst.transforms.V2V3ToIdeal.v2ref
```

and for ``fits_generator``:
```
/Users/bushouse/JWST/softw/jwst/docs/jwst/fits_generator/template.rst:132: WARNING: undefined label: generators (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/fits_generator/template.rst:132: WARNING: undefined label: generators (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/fits_generator/template.rst:352: WARNING: undefined label: verifiers (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/fits_generator/template.rst:352: WARNING: undefined label: verifiers (if the link has no caption the label must precede a section header)
```

and for ``outlier_detection``:
```
/Users/bushouse/JWST/softw/jwst/docs/jwst/outlier_detection/outlier_detection_ifu.rst:11: WARNING: undefined label: outlier-detection-image (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/outlier_detection/outlier_detection_ifu.rst:11: WARNING: undefined label: outlier-detection-image (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/outlier_detection/outlier_detection_spec.rst:11: WARNING: undefined label: outlier-detection-image (if the link has no caption the label must precede a section header)
/Users/bushouse/JWST/softw/jwst/docs/jwst/outlier_detection/outlier_detection_spec.rst:11: WARNING: undefined label: outlier-detection-image (if the link has no caption the label must precede a section header)
```

**Comment by stscieisenhamer** [comment]
For association-related stuff, this should be documented in #168. Yea, note the low issue number.

**Comment by stscirij** [comment]
I fixed fits_generator sections in #2290

**Comment by stsci-hack** [comment]
I believe I have fixed the outlier_detection warnings in #2291.

**Comment by hbushouse** [comment]
Three of the four cases have been fixed (transforms, fits_generator, outlier_detection) and the associations-related cases are being tracked in #168. Closing this issue.
